tidy3d/components/structure.py

```diff
@@ -197,11 +197,14 @@ def to_gdstk(
 
         if isinstance(self.medium, AbstractCustomMedium):
             axis, _ = self.geometry.parse_xyz_kwargs(x=x, y=y, z=z)
+            bb_min, bb_max = self.geometry.bounds
 
             eps, _, _ = self.medium.eps_dataarray_freq(frequency=frequency)
-            scale = min(np.diff(eps.x).min(), np.diff(eps.y).min(), np.diff(eps.z).min())
+            scale = max(b - a for a, b in zip(bb_min, bb_max))
+            for coord in (eps.x, eps.y, eps.z):
+                if len(coord) > 1:
+                    scale = min(scale, np.diff(coord).min())
```

Comment:
The pixel size was already taken into account before. The scale is the minimal distance between custom medium points with respect to each direction (x, y, z). The problem was that, if any of the directions had a single point, `np.diff` returned an empty array and the `min` function would raise an error. Now we initialize the scale with the largest possible value (from the bounding box) and, for each direction with at least 2 coordinates, keep only the minimum. Does it make sense to you? If so, I'll add those comments.

Reply:
oh I see, yea if you could just add a comment in the source and maybe make the changelog item more specific to that effect, it would help a lot, thanks for explaining.

```
- Properly handle length sale in exporting custom medium structures to GDS when the geometry size is very small along some dimensions.
```
or something